core-api/src/main/java/com/optimizely/ab/odp/ODPEventManager.java

```diff
@@ -107,7 +107,7 @@ public void identifyUser(@Nullable String vuid, @Nullable String userId) {
         if (vuid != null) {
             identifiers.put(ODPUserKey.VUID.getKeyString(), vuid);
         }
-        if (userId != null) {
+        if (userId != null && !isVuid(userId)) {
             identifiers.put(ODPUserKey.FS_USER_ID.getKeyString(), userId);
         }
```

Comment:
This is a great catch! We also need to consider to add vuid as userId (android-sdk passes it as a normal userId). 
I'll add new test cases into android-sdk.

```suggestion
        if (userId != null) {
            if (isVuid(userId)) {
                        identifiers.put(ODPUserKey.VUID.getKeyString(), userId);
            } else {
                        identifiers.put(ODPUserKey.FS_USER_ID.getKeyString(), userId);
            }
        }
```